pkg/build/builder/source.go

```diff
@@ -252,7 +252,16 @@ func extractGitSource(gitClient GitClient, gitSource *api.GitBuildSource, revisi
 		}
 
 		if err := gitClient.Checkout(dir, commit); err != nil {
-			return true, err
+			//TODO - do we want to parse the error string for the text that accompanies a clone / checkout of a PR ref, such as:   pathspec 'refs/pull/1/head' did not match any file(s) known to git
+			err = gitClient.Fetch(dir, gitSource.URI, commit)
```

Comment:
it's probably worth adding some level 4 logging to this code path so we know when we've gone down it successfully (or if we ever go down it incorrectly).